src/Data/Common/UrlState/useUrlStateWithCurrentPage.ts

```diff
@@ -0,0 +1,50 @@
+import { isEqual, isArray } from "lodash";
+import { provide, Location, Replace, StateConfig, Update } from "./helpers";
+import { handleUrlState } from "./useUrlState";
+
+export const useUrlStateWithCurrentPage = provide(
+  handleUrlStateWithCurrentPage,
+);
+const valueIsValid = (value: unknown): value is string[] => {
+  return isArray(value) && value.every((param) => typeof param === "string");
+};
+
+const from = (value: string[]): CurrentPage => {
+  return { kind: "CurrentPage", value: valueIsValid(value) ? value : [] };
+};
+const serialize = (currentPage: CurrentPage): string[] => {
```

Comment:
serialize is for me more than just returning the currentPage.value.

Reply:
rearranged a little bit, but still that's the most of serialisation  that could be done inside